File: 1. Two Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = {};
    for (let i = 0; i < nums.length; i++) {
        const num = target - nums[i];
        if (num in map) {
            return [i, map[num]];
        }
        map[nums[i]] = i;
    }
};

File: 1. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_index_map = {}

        for i, num in enumerate(nums):
            num2 = target - num
            if num2 in num_index_map:
                return [num_index_map[num2], i]
            num_index_map[num] = i

        return []